// pg-test-project/src/components/PaymentInitModal/PaymentInitModal.stories.tsx
import React from "react";
import { Meta } from "@storybook/react/types-6-0";
import { Story } from "@storybook/react";
import PaymentModal from "./ContactInformation";

import Pg, { PgModalProps } from "./Pg";

export default {
  title: "Components/PaymentInitModal",
  component: PaymentModal,
  argTypes: {
  },
} as Meta;

// Create a master template for mapping args to render the Button component
const PgNew: Story<PgModalProps> = (args) => <Pg {...args} />;
export const pg = PgNew.bind({});
pg.args = {
  isOpen: true,
  authkey: "kaY9AIhuJZNvKGp2",
  authiv: "YN2v8qQcU3rGfA1y",
  payerName: "",
  payerEmail: "akshaycyber96@gmail.com",
  payerMobile: "7018271479",
  clientTxnId: "TESTING090922085442157",
  payerAddress: "vpo dhaned",
  amount: "500",
  clientCode: "TM001",
  transUserName: "rajiv.moti_336",
  transUserPassword: "RIADA_SP336",
  amountType: "INR",
  udf1: "",
  udf2: "",
  udf3: "",
  udf4: "",
  udf5: "",
  udf6: "",
  udf7: "",
  udf8: "",
  udf9: "",
  udf10: "",
  udf11: "",
  udf12: "",
  udf13: "",
  udf14: "",
  udf15: "",
  udf16: "",
  udf17: "",
  udf18: "",
  udf19: "",
  udf20: "",
  channelId: "npm",
  programId: "",
  mcc: "",
  env: "dev"
}
